From c8a34bf2123784d888840dc160f7e0ce30e5aaef Mon Sep 17 00:00:00 2001
From: Serge BARANDAO <serge.barandao-ext@st.com>
Date: Mon, 3 Jun 2024 15:50:31 +0200
Subject: [PATCH 611/954] feat(fdts): add ospi1 node for stm32mp21

Add ospi1 node for stm32mp21.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b7c30aeb5a3d7894e81cdabc79a7c2312bfbdaef feat(fdts): add ospi1 node for stm32mp21]
Change-Id: Ib39e3a417b23b7d7e19fdcea2ac8c1fb06c4f318
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/384302
Domain-Review: Serge BARANDAO <serge.barandao-ext@st.com>
Tested-by: Serge BARANDAO <serge.barandao-ext@st.com>
Reviewed-by: Serge BARANDAO <serge.barandao-ext@st.com>
---
 fdts/stm32mp21-bl2.dtsi  | 5 +++++
 fdts/stm32mp21-bl31.dtsi | 1 +
 fdts/stm32mp211.dtsi     | 8 ++++++++
 3 files changed, 14 insertions(+)

diff --git a/fdts/stm32mp21-bl2.dtsi b/fdts/stm32mp21-bl2.dtsi
index d62426e614..ec378cdf94 100644
--- a/fdts/stm32mp21-bl2.dtsi
+++ b/fdts/stm32mp21-bl2.dtsi
@@ -7,6 +7,11 @@
 
 / {
 	soc@0 {
+#if !STM32MP_SPI_NAND && !STM32MP_SPI_NOR
+		bus@42080000 {
+			/delete-node/ spi@40430000;
+		};
+#endif
 #if !STM32MP_RAW_NAND
 		/delete-node/ memory-controller@48200000;
 #endif
diff --git a/fdts/stm32mp21-bl31.dtsi b/fdts/stm32mp21-bl31.dtsi
index 5f45a21d05..4790bd1a5d 100644
--- a/fdts/stm32mp21-bl31.dtsi
+++ b/fdts/stm32mp21-bl31.dtsi
@@ -8,6 +8,7 @@
 		/delete-node/ memory-controller@48200000;
 
 		bus@42080000 {
+			/delete-node/ spi@40430000;
 			/delete-node/ mmc@48220000;
 			/delete-node/ mmc@48230000;
 		};
diff --git a/fdts/stm32mp211.dtsi b/fdts/stm32mp211.dtsi
index 7a675577b8..e8b1656f5f 100644
--- a/fdts/stm32mp211.dtsi
+++ b/fdts/stm32mp211.dtsi
@@ -205,6 +205,14 @@
 				status = "disabled";
 			};
 
+			ospi1: spi@40430000 {
+				compatible = "st,stm32mp25-omi";
+				reg = <0x40430000 0x400>;
+				clocks = <&rcc CK_KER_OSPI1>;
+				resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+				status = "disabled";
+			};
+
 			rng2: rng@42020000 {
 				compatible = "st,stm32mp13-rng";
 				reg = <0x42020000 0x400>;
-- 
2.53.0

